service/matching/liveness.go

```diff
@@ -25,103 +25,49 @@
 package matching
 
 import (
-	"sync"
 	"sync/atomic"
 	"time"
 
-	"go.temporal.io/server/common"
-	"go.temporal.io/server/common/clock"
+	"github.com/jonboulle/clockwork"
 )
 
 type (
 	liveness struct {
-		status     int32
-		timeSource clock.TimeSource
-		ttl        time.Duration
-		// internal shutdown channel
-		shutdownChan chan struct{}
-
-		// broadcast shutdown functions
-		broadcastShutdownFn func()
+		clock  clockwork.Clock
+		ttl    func() time.Duration
+		onIdle func()
+		timer  atomic.Value
+	}
 
-		sync.Mutex
-		lastEventTime time.Time
+	timerWrapper struct {
```

Comment:
I thought so too, but it doesn't work due to the nil in Stop. You get
```
panic: sync/atomic: swap of nil value into Value
```
on Stop even if I try to pass an interface value of the correct type. I think it's because it comes in as an interface{} so any other interface type gets dropped? If you want to give it a try, please do